test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -29,6 +28,26 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; JRubyPuppet Test util functions
 
+(schema/defn ^:always-validate
+create-mock-pool-instance :- JRubyInstance
```

Comment:
funny you should mention. I initially did mock `get-initialize-pool-instance-fn` instead of `create-pool-instance`, and it led us down a rabbit hole for 2 days and took 3 developers to figure out why a test was failing. In the test `class-info-updated-after-cache-flush-during-prior-request`, there is a timeout of 10 seconds, which would cause the test to fail, but only when run with a lot of other tests. It would never fail when run by itself. Chris hypothesized it must have something to do with garbage collection, but we never really figured out why. Either way, after that we were so scarred we decided to just fix it the quickest way possible